internal/controllers/compute/machine_ephemeralvolume_controller.go

```diff
@@ -71,19 +72,29 @@ func (r *MachineEphemeralVolumeReconciler) ephemeralMachineVolumeByName(machine
 				Labels:      ephemeral.VolumeTemplate.Labels,
 				Annotations: maps.Clone(ephemeral.VolumeTemplate.Annotations),
 			},
-			Spec: ephemeral.VolumeTemplate.Spec,
+			Spec: ephemeral.VolumeTemplate.Spec.VolumeSpec,
 		}
 		annotations.SetDefaultEphemeralManagedBy(volume)
-		_ = ctrl.SetControllerReference(machine, volume, r.Scheme())
+		if machineVolume.Ephemeral.VolumeTemplate.Spec.ReclaimPolicy != storagev1alpha1.Retain {
+			_ = ctrl.SetControllerReference(machine, volume, r.Scheme())
+		}
```

Comment:
if we only set the controller reference in case its `Reclaim` we are loosing the association for reconciliation. I am unsure if we can utilize https://kubernetes.io/docs/tasks/administer-cluster/use-cascading-deletion/#set-orphan-deletion-policy because this is set during deletion of the controlling object, in this case when the `Machine` is deleted and has to be set by the caller.

We still have the `.spec.ClaimRef` as an association between `Machine` and `Volume` that can be sufficient for reconciliation means. Lets discuss in the sync call how to handle this.

Reply:
As discussed when we set ownerRef, machine deletion also trigger setting the deletion timestamp in the owned objects, so we will handle it without adding ownerRef if reclaimPolicy is set to Retain.